create.go

```diff
@@ -217,45 +220,79 @@ func (r *Resource) computeCreateEventPatches(ctx context.Context, obj interface{
 	// from the NodeConfig CR status. This is not possible right now because the
 	// NodeConfig CRs are still used for draining by older tenant clusters.
 	{
-		k8sClient := r.k8sClient.K8sClient()
-
-		o := metav1.ListOptions{}
-		list, err2 := k8sClient.CoreV1().Nodes().List(o)
-		if tenant.IsAPINotAvailable(err2) {
-			// fall through
-		} else if err2 != nil {
-			return nil, microerror.Mask(err2)
-		} else {
-			var nodes []providerv1alpha1.StatusClusterNode
-
-			for _, node := range list.Items {
-				l := node.GetLabels()
-				n := node.GetName()
-
-				labelProvider := "giantswarm.io/provider"
-				p, ok := l[labelProvider]
-				if !ok {
-					return nil, microerror.Maskf(missingLabelError, labelProvider)
-				}
-				labelVersion := p + "-operator.giantswarm.io/version"
-				v, ok := l[labelVersion]
-				if !ok {
-					return nil, microerror.Maskf(missingLabelError, labelVersion)
-				}
+		var k8sClient kubernetes.Interface
+		{
+			r.logger.LogCtx(ctx, "level", "debug", "message", "creating Kubernetes client for tenant cluster")
+
+			i, err := r.clusterIDFunc(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			e, err := r.clusterEndpointFunc(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			restConfig, err := r.tenantCluster.NewRestConfig(ctx, i, e)
```

Comment:
`create.go` is the same as two commits before the current master. Only `tenantCluster` here use a different function.